storm-server/src/main/java/org/apache/storm/container/cgroup/CgroupManager.java

```diff
@@ -151,22 +161,40 @@ public void reserveResourcesForWorker(String workerId, Map<String, Number> resou
             }
         }
 
-        if (totalMem != null) {
-            MemoryCore memCore = (MemoryCore) workerGroup.getCores().get(SubSystemType.memory);
-            try {
-                memCore.setPhysicalUsageLimit(Long.valueOf(totalMem.longValue() * 1024 * 1024));
-            } catch (IOException e) {
-                throw new RuntimeException("Cannot set memory.limit_in_bytes! Exception: ", e);
+        if ((boolean) this.conf.get(DaemonConfig.STORM_CGROUP_MEMORY_ENFORCEMENT_ENABLE)) {
+            if (totalMem != null) {
+                int cgroupMem =
+                    (int)
+                        (Math.ceil(
+                            ObjectReader.getDouble(
+                                this.conf.get(DaemonConfig.STORM_CGROUP_MEMORY_LIMIT_TOLERANCE_MARGIN_MB),
```

Comment:
I'm wondering if having defaults here is necessary? Isn't defaults.yaml always expected to be present?